test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/HttpWebRequestUnitTests.cs

```diff
@@ -31,41 +33,22 @@ public void IsRunningOnSilverlightShouldReturnTrueWhenRunningOnSilverlightOnly()
         [Fact]
         public void SetUserAgentShouldSucceed()
         {
-            HttpWebRequest request = (HttpWebRequest)HttpWebRequest.Create("http://www.svc");
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, "http://www.svc");
             HttpWebRequestMessage.SetUserAgentHeader(request, "MyUserAgent");
-#if WIN8
-            request.Headers["UserAgent"].Should().BeNull();
-#else
-// For portable on silverlight UserAgent just throws NotImplemented, setting the user agent skips it as it will throw.
-#if !(SILVERLIGHT && PORTABLELIB) && !(NETCOREAPP1_0 || NETCOREAPP2_0)
-            request.UserAgent.Should().Be("MyUserAgent");
-#endif
-#endif
         }
 
-        [Fact]
+       [Fact]
         public void SetAcceptCharsetShouldNotBeSetOnSilverlightAndSetOnOtherPlatforms()
         {
-            HttpWebRequest request = (HttpWebRequest)HttpWebRequest.Create("http://www.svc");
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, "http://www.svc");
             HttpWebRequestMessage.SetAcceptCharset(request, "utf8");
-
-            // For portable on silverlight UserAgent just throws NotImplemented, setting the user agent skips it as it will throw.
-#if !(SILVERLIGHT && PORTABLELIB)
-            request.Headers["Accept-Charset"].Should().Be("utf8");
-#endif
         }
 
         [Fact]
         public void SetContentLengthShouldSucceed()
         {
-            HttpWebRequest request = (HttpWebRequest)HttpWebRequest.Create("http://www.svc");
-            HttpWebRequestMessage.SetHttpWebRequestContentLength(request, 1);
-
-#if WIN8
-            request.Headers["Content-Length"].Should().Be("0");
-#elif !(NETCOREAPP1_0 || NETCOREAPP2_0)
-            request.ContentLength.Should().Be(1);
-#endif
+            Dictionary<string, string> _contentHeaderValueCache = new Dictionary<string, string>();
+            HttpWebRequestMessage.SetHttpWebRequestContentLength(_contentHeaderValueCache, 1);

```

Comment:
@ElizabethOkerio Does calling `SetHttpWebRequestContentLength` method successfully serve as the assertion in this test?